src/Dumplie/Infrastructure/Doctrine/Dbal/Metadata/TypeRegistry.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Infrastructure\Doctrine\Dbal\Metadata;
+
+use Doctrine\DBAL\Schema\Table;
+use Dumplie\Application\Metadata\Schema\FieldDefinition;
+use Dumplie\Infrastructure\Doctrine\Dbal\Metadata\Field\TextMapping;
+
+final class TypeRegistry
+{
+    /**
+     * @var TypeMapping[]
+     */
+    private $mapping;
+
+    public function __construct(array $typeMappings = [])
+    {
+        $this->register(new TextMapping());
```

Comment:
yea this instantiation sounds like a good idea but with static constructor you can also avoid instantiation of objects that you are going to replace, you can simply use default constructor and pass only those types that you need